akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala

```diff
@@ -385,27 +390,49 @@ private[http] object HttpServerBluePrint {
                 oneHundredContinueResponsePending = true
                 r.copy(createEntity = with100ContinueTrigger(r.createEntity))
               } else r
+              lastRequestOnConnectionSummary = s"${r.method.value} ${r.uri.path}"
               push(requestPrepOut, rs)
+
             case MessageEnd ⇒
               messageEndPending = false
               push(requestPrepOut, MessageEnd)
-            case MessageStartError(status, info) ⇒ finishWithIllegalRequestError(status, info)
+
+            case MessageStartError(status, info) ⇒
+              finishWithIllegalRequestError(status, info)
+
             case x: EntityStreamError if messageEndPending && openRequests.isEmpty ⇒
               // client terminated the connection after receiving an early response to 100-continue
               completeStage()
+
             case x ⇒
               push(requestPrepOut, x)
           }
-        override def onUpstreamFinish() =
+
+        override def onUpstreamFinish =
           if (openRequests.isEmpty) completeStage()
           else complete(requestPrepOut)
+
+        override def onUpstreamFailure(ex: Throwable): Unit = {
+          ex match {
+            case t: TimeoutException ⇒
+              // FIXME: this must be `akka.stream.impl.io.TcpIdleTimeoutException` instead! Then improve below exception:
+              log.debug(
+                "Timeout exception occured while replying to request: [{}], shutting down connection. Cause: {}",
```

Comment:
I totally agree with the result to achieve. But as you say the improvement of the message must be done in akka. The only thing we can do here is to add context.

Once the akka message says `TCP connection was closed because no data was transferred for the configured idle-timeout of 10 seconds.`, I don't think we need to add here anything more than the generic prefix `HTTP connection was closed during handling of the request GET /bla because of a connection error: ...`.

The total error message would then be:

`HTTP connection was closed during handling of the request GET /bla because of a connection error: TCP connection was closed because no data was transferred for the configured idle-timeout of 10 seconds.`

What would you add to that?